microsoft-edge/web-platform/release-notes/135.md

```diff
@@ -0,0 +1,462 @@
+
+
+<!-- ---------- -->
+###### HSTS tracking prevention
+
+Mitigates user tracking by third-parties via the HSTS<!-- todo: define HSTS --> cache.
+
+This feature only allows HSTS upgrades for top-level navigations, and blocks HSTS upgrades for sub-resource requests.  Blocking such an HSTS upgrade makes it infeasible for third-party sites to use the HSTS cache in order to track a user across the web.
+
+
+<!-- ---------- -->
+###### `sourceElement` property for `NavigateEvent`
+
+When a page navigation is initiated by an element, such as when the user clicks a link or submits a form, the resulting `NavigateEvent` now has a `sourceElement` property that points to the initiating element.
+
+
+<!-- ---------- -->
+###### PerformanceNavigationTiming API NotRestoredReasons name change
+
+The NotRestoredReasons API is part of the PerformanceNavigationTiming API. This change updates the names of the reasons so that they match the specification.
+
+List of changes:
+
+| Old name | New name |
+| --- | --- |
+| `extension-messaging`, `pending-permission-request`, `not-main-frame`, `frame-navigating`, `timeout`, `cookie-disabled`, `cookie-removed`, `printing`, `web-database`, `injected-javascript`, and `injected-stylesheet` | `masked` |
+| `response-method-not-get` | `request-method-not-get` |
+| `serviceworker-version-activation` | `serviceworker-version-activated` |
+| `serviceworker-added-after-bfcache` | `serviceworker-added` |
+| `serviceworker-unregistration` | `serviceworker-unregistered` |
+| `unload-handler` | `unload-listener` |
+| `session-restored` | `null` |
+
+
+<!-- ---------- -->
+###### Observable API
+
+Observables are a popular reactive-programming paradigm to handle an asynchronous stream of push-based events.  Observables can be thought of as Promises but for multiple events, and aim to do what Promises did for callbacks and nesting.  That is, Observables allow ergonomic event handling, by providing an `Observable` object that represents the asynchronous flow of events.
+
+You can "subscribe" to this object to receive events as they come in, and call any of its operators or combinators to declaratively describe the flow of transformations through which events go.  This is in contrast with the _imperative_ version, which often requires complicated nesting with approaches such as `addEventListener()`.  For more on this, see the examples in the explainer.<!-- todo: link to the explainer.  if that's the next link, clarify. -->
+
+See the [W3C Observable specification](https://wicg.github.io/observable/).
+
+
+<!-- ---------- -->
+###### Service worker client URL now ignores `history.pushState` changes
+
+The `Client.url` property of a service worker now ignores document URL changes that are made by the `history.pushState()` method and other similar history APIs.  The `Client.url` property is intended to be the creation URL of the HTML document,<!-- todo: check meaning of comma vs no comma --> which ignores such changes.
```

Comment:
confirm whether to use comma.  or clarify.  read it w/ comma & without; which one gives clearest meaning?